src/Data/Ord/Linear.hs

```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE MonoLocalBinds #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+
+module Data.Ord.Linear
+  ( Ord(..)
+  )
+  where
+
+import Data.Eq.Linear
+import qualified Prelude as Unrestricted
+import Prelude (Bool(..))
+import Data.Ord (Ordering(..))
+import qualified Unsafe.Linear as Unsafe
+
+class Eq a => Ord a where
+  {-# MINIMAL compare | (<=) #-}
+  compare :: a #-> a #-> Ordering
+  compare = Unsafe.toLinear2 compare'
+    where
+      compare' :: Ord a => a -> a -> Ordering
+      compare' x y
+        | x == y = EQ
+        | x <= y = LT
+        | True = GT
```

Comment:
If you need `Unsafe.toLinear`, then it's probably unsafe :wink: . 

The conclusion is that there is no linear definition of `compare` based on just `(==)` and `(<=)`, and that's fine.
```suggestion
```